### DISCUSSION TITLE
Does `gix` static link TLS library?

### DISCUSSION THREAD

**Discussion by KSXGitHub** [GitoxideLabs/gitoxide#776] [Q&A] [answered]
I'm planning to migrate some of my existing tools from  the `git2` crate (`libgit2`) to the `gix` crate because dynamic dependencies have been causing headaches.

My question is: 
* What kind of TLS library does `gix` uses?
* Is it statically linked?
* How to enable/disable this feature?

**Comment by Byron** [comment, marked as answer]
`gix` supports [a couple of HTTP transport implementations](https://docs.rs/gix/0.41.0/gix/#mutually-exclusive-network-client), with one being `curl` and another being `reqwest`. `curl` supports [different TLS implementations](https://docs.rs/crate/curl/0.4.44/features), and I never tried to change the default. 

It should be possible to pull in `curl` as a direct dependency and alter its feature flags, to affect the entire dependency tree, allowing to configure the TLS implementation that is used by the application.

I recommend experimentation with the options outlined above, maybe that works already, or  concrete idea of what needs improvement arises from that.
